extensions/profiler/client/src/dialogs/profilerCreateSessionDialog.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+import * as nls from 'vscode-nls';
+import * as sqlops from 'sqlops';
+import * as vscode from 'vscode';
+import { CreateSessionData } from '../data/createSessionData';
+
+const localize = nls.loadMessageBundle();
+
+export class CreateSessionDialog {
+
+	// Top level
+	private readonly DialogTitle: string = localize('createSessionDialog.newSession', 'New Session');
+	private readonly CancelButtonText: string = localize('createSessionDialog.cancel', 'Cancel');
+	private readonly CreateButtonText: string = localize('createSessionDialog.create', 'Create');
+	private readonly DialogTitleText: string = localize('createSessionDialog.title', 'Create New Profiler Session');
+
+	// UI Components
+	private dialog: sqlops.window.modelviewdialog.Dialog;
+	private templatesBox: sqlops.DropDownComponent;
+	private sessionNameBox: sqlops.InputBoxComponent;
+
+	private model: CreateSessionData;
+
+	private _onSuccess: vscode.EventEmitter<CreateSessionData> = new vscode.EventEmitter<CreateSessionData>();
+	public readonly onSuccess: vscode.Event<CreateSessionData> = this._onSuccess.event;
+
+
+	constructor(ownerUri: string, templates: Array<sqlops.ProfilerSessionTemplate>) {
+		if (typeof (templates) === 'undefined' || templates === null) {
+			throw new Error(localize('createSessionDialog.templatesInvalid', "Invalid templates list, cannot open dialog"));
+		}
+		if (typeof (ownerUri) === 'undefined' || ownerUri === null) {
+			throw new Error(localize('createSessionDialog.dialogOwnerInvalid', "Invalid dialog owner, cannot open dialog"));
+		}
+		this.model = new CreateSessionData(ownerUri, templates);
+	}
+
+	public async showDialog() {
+		this.dialog = sqlops.window.modelviewdialog.createDialog(this.DialogTitle);
+		this.initializeContent();
+		this.dialog.okButton.onClick(async () => await this.execute());
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
```

Comment:
This and the handler above don't need to use async/await since they don't actually need to wait on the handler execution